### DISCUSSION TITLE
Iterating through site collections using Connect-PnPOnline

### DISCUSSION THREAD

**Discussion by benjacernuda** [pnp/powershell#1492] [Q&A]
I would need to iterate through many site collections in order to install a new app. the user I'm using has MFA enabled. 

Is there a way or parameter to use connect-pnponline without introduce the password for each site collecton? 

I have tried to use -SPOManagementShell but it seems is not working with the new PnP.Powershell

Many thanks in advance!

**Comment by get-itips** [comment]
I need the same, trying to iterate to every OneDrive (personal/) and I want to avoid creating a connection to each one, is this possible currently in PnP PowerShell?

`Connect-PnpOnline` I already tried connecting against https://tenant-my.sharepoint.com and https://tenant-my.sharepoint.com/personal, but I can't go "down" and get specific SC under those, with this same connection.
